solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -195,6 +196,32 @@ pub mod solana_ibc {
         #[cfg(not(feature = "mocks"))]
         panic!("This method is only for mocks");
     }
+
+    /// Should be called after setting up client, connection and channels.
+    pub fn send_packet(ctx: Context<Deliver>, packet: Packet) -> Result<()> {
+        let private: &mut storage::PrivateStorage = &mut ctx.accounts.storage;
+        let provable = storage::get_provable_from(&ctx.accounts.trie, "trie")?;
+        let packets = &mut ctx.accounts.packets;
+        let host_head = host::Head::get()?;
+
+        // Before anything else, try generating a new guest block.  However, if
+        // that fails it’s not an error condition.  We do this at the beginning
+        // of any request.
+        ctx.accounts.chain.maybe_generate_block(&provable, Some(host_head))?;
+
+        let mut store = storage::IbcStorage::new(storage::IbcStorageInner {
+            private,
+            provable,
+            packets,
+            accounts: Vec::new(), // We are not doing any transfers so no point of having accounts
+            host_head,
+        });
+
+        ibc::core::send_packet(&mut store, packet)
+            .map_err(error::Error::ContextError)
+            .map_err(|err| error!((&err)))?;
+        Ok(())
```

Comment:
```suggestion
            .map_err(|err| error!((&err)))
```
should work here, no?

Reply:
Yes yes, the above would work.

Reply:
Nvm, saw that its already been added to master with #128 .